### ISSUE TITLE
[Question] Not see fps improvements comparing sbx ppo with sb3 ppo

### ISSUE THREAD

**Issue by zzhixin** [araffin/sbx#70]
### Question
I tested sbx ppo and sb3 ppo on gym env `Hopper-v5`. The results shows that neither sbx on cpu or sbx on cuda is significantly faster than sb3 on cpu. 

![Image](https://github.com/user-attachments/assets/c3ab7967-eb8d-4f7d-b101-52e524e583b9)

![Image](https://github.com/user-attachments/assets/82ce5586-c2a5-4210-a620-3bfa0f5454fc)

My question are:
1. Is the result expected? if it is what is benefit to use sbx ppo? if not where did I implement wrong?
2. Is the performance improvements claimed [here](https://stable-baselines3.readthedocs.io/en/master/guide/sbx.html#:~:text=It%20provides%20a%20minimal%20number%20of%20features%20compared%20to%20SB3%20but%20can%20be%20much%20faster%20(up%20to%2020x%20times!)%3A) still true in 2025?

### Additional context

The hyparameters of all runs are kept consistent as below:

```
batch_size:64
device:"cuda"
env_id:"hopper"
learning_rate:0.0001
n_envs:16
n_epochs:10
n_steps:128
seed:0
total_timesteps:1,000,000
```

System hardware:

item | value
-- | --
CPU count | 8
Logical CPU count | 16
GPU count | 1
GPU type | NVIDIA GeForce RTX 3080 Ti

Here is the wandb report: https://wandb.ai/zhixin/jaxrl/reports/performance-compare-sbx-vs-sb3-ppo--VmlldzoxMjU0ODY2NQ

Code can be found [here](https://wandb.ai/zhixin/jaxrl/runs/53zwrxdh/code)

Run script:
```bash
# sbx gym cpu
python -O sb3_learn.py --sbx --seed 0 --n_envs 16 --n_steps 128 --learning_rate 0.0001 --batch-size 64 --total_timesteps 1000000

# sbx gym cuda
python -O sb3_learn.py --sbx --device cuda --seed 0 --n_envs 16 --n_steps 128 --learning_rate 0.0001 --batch-size 64 --total_timesteps 1000000


# sb3 gym cpu
python -O sb3_learn.py --seed 0 --n_envs 16 --n_steps 128 --learning_rate 0.0001 --batch-size 64 --total_timesteps 1000000
```

**Comment by zzhixin** [comment]
After research, I found the reason is I test sbx on wsl2 system rather than native Linux system. JAX official tutorial says JAX on wsl is [experimental](https://docs.jax.dev/en/latest/installation.html#supported-platforms).

After switching to Linux (Ubuntu 18.04) on same hardware, sbx shows significant performance improvement than sb3. Below is the fps of PPO on env Hopper with sbx cuda, sbx cpu and sb3 cpu.

![Image](https://github.com/user-attachments/assets/4dcf6b36-446f-40d9-a615-205193e03a98)

**Comment by araffin** [comment]
Hello,
good to hear that you found the issue =)
The perf boost reported is mostly for SAC (and when combining multiple gradients steps, from 10x to 20x faster), with PPO, there should be a boost but not as much as SAC.
Also, PPO on CPU will be faster than PPO on GPU unless you have a big network (we have different issues about that in SB3 repo and SB3 PPO also outputs a warning).
